jobs/otel-collector-windows/templates/config.yml.erb

```diff
@@ -1,35 +1,50 @@
 <%=
 def config
   @config ||= begin
-    cfg = retrieve_config
+    cfg = retrieve_property('config')
+    secrets = retrieve_property('secrets').to_h { |x| [x['name'], { 'cert' => x['cert'], 'key' => x['key'], 'ca' => x['ca'], 'secret' => x['secret'] }] }
+    used_secrets = {}
```

Comment:
seems like this could be cleaner as an array?

Reply:
We used a Hash on purpose for faster retrieval in `check_for_unused_secrets!` while checking for presence.

Is the cleanliness worth the slightly worse performance? Or is there a way to find elements in Ruby Arrays that is equivalently fast to Hash retrieval?

Reply:
I briefly looked this up, looks like using Set might be the cleanest way to do this.